geomesa-hbase/geomesa-hbase-spark/src/main/scala/org/locationtech/geomesa/spark/hbase/HBaseSpatialRDDProvider.scala

```diff
@@ -0,0 +1,112 @@
+
+  def rdd(conf: Configuration,
+          sc: SparkContext,
+          dsParams: Map[String, String],
+          origQuery: Query): RDD[SimpleFeature] = {
+    val ds = DataStoreFinder.getDataStore(dsParams).asInstanceOf[HBaseDataStore]
+
+    // get the query plan to set up the iterators, ranges, etc
+    lazy val sft = ds.getSchema(origQuery.getTypeName)
+    lazy val qp = ds.getQueryPlan(origQuery).head
+
+    if (ds == null || sft == null || qp.isInstanceOf[EmptyPlan]) {
+      sc.emptyRDD[SimpleFeature]
+    } else {
+      import org.locationtech.geomesa.index.conf.QueryHints._
+
+      val query = ds.queryPlanner.configureQuery(origQuery, sft)
+      GeoMesaConfigurator.setSerialization(conf)
+      query.getHints.getTransformSchema.foreach(GeoMesaConfigurator.setTransformSchema(conf, _))
+      GeoMesaConfigurator.setTable(conf, qp.table.getNameAsString)
+      GeoMesaConfigurator.setDataStoreInParams(conf, dsParams)
+      GeoMesaConfigurator.setFeatureType(conf, sft.getTypeName)
+      qp.filter.secondary.foreach { f => GeoMesaConfigurator.setFilter(conf, ECQL.toCQL(f)) }
+      val scans = qp.ranges.map { s =>
+        val scan = s
+        // need to set the table name in each scan
+        scan.setAttribute(Scan.SCAN_ATTRIBUTES_TABLE_NAME, qp.table.getName)
+        convertScanToString(scan)
+      }
+      conf.setStrings(MultiTableInputFormat.SCANS, scans: _*)
+
+      sc.newAPIHadoopRDD(conf, classOf[GeoMesaHBaseInputFormat], classOf[Text], classOf[SimpleFeature]).map(U => U._2)
+    }
+  }
+
+
+  private def convertScanToString(scan: org.apache.hadoop.hbase.client.Query): String = scan match {
+    case g: Get =>
+      val proto = ProtobufUtil.toGet(g)
+      Base64.encodeBytes(proto.toByteArray)
+
+    case s: Scan =>
+      val proto = ProtobufUtil.toScan(s)
+      Base64.encodeBytes(proto.toByteArray)
+  }
+
+
+  /**
+    * Writes this RDD to a GeoMesa table.
+    * The type must exist in the data store, and all of the features in the RDD must be of this type.
+    *
+    * @param rdd
+    * @param writeDataStoreParams
+    * @param writeTypeName
+    */
+  def save(rdd: RDD[SimpleFeature], writeDataStoreParams: Map[String, String], writeTypeName: String): Unit = {
+    val ds = DataStoreFinder.getDataStore(writeDataStoreParams).asInstanceOf[HBaseDataStore]
+    try {
+      require(ds.getSchema(writeTypeName) != null,
+        "Feature type must exist before calling save.  Call createSchema on the DataStore first.")
+    } finally {
+      ds.dispose()
+    }
+
+    rdd.foreachPartition { iter =>
+      val ds = DataStoreFinder.getDataStore(writeDataStoreParams).asInstanceOf[HBaseDataStore]
+      val featureWriter = ds.getFeatureWriterAppend(writeTypeName, Transaction.AUTO_COMMIT)
+      try {
+        iter.foreach { rawFeature =>
+          FeatureUtils.copyToWriter(featureWriter, rawFeature, overrideFid = true)
+          featureWriter.write()
+        }
+      } finally {
+        IOUtils.closeQuietly(featureWriter)
```

Comment:
i've run into problems with IOUtils.closeQuietly since i think it only catches IOExceptions and accumulo/hbase can throw their own things. So i started using ```try { featureWriter.close() } ``` with a catch with noop